codec/dagjson/marshal.go

```diff
@@ -31,21 +32,32 @@ func Marshal(n ipld.Node, sink shared.TokenSink, allowLinks bool) error {
 		if _, err := sink.Step(&tk); err != nil {
 			return err
 		}
-		// Emit map contents (and recurse).
+		// Collect map entries, then sort by key
+		type entry struct {
```

Comment:
is declaring a struct in the middle of this naughty, or OK? can I get away without a struct somehow?